core/src/main/java/org/mule/runtime/core/internal/profiling/DefaultProfilingService.java

```diff
@@ -193,4 +197,19 @@ private String getScope() {
       return getArtifactId(muleContext);
     }
   }
+
+  @Override
+  public ExportedSpanCapturer getExportedSpanCapturer() {
+    return eventTracer.getSpanCapturer();
+  }
+
+  @Override
+  public void startComponentSpan(CoreEvent event, Component component) {
```

Comment:
Let's add a TODO here in order to tackle the FlowRunner limitations and be able to remove this "testing specific" logic.